phpunit/class-wp-theme-json-test.php

```diff
@@ -406,6 +406,14 @@ public function test_merge_incoming_data() {
 					'custom'         => true,
 					'customGradient' => true,
 					'palette'        => array(
+						array(
+							'slug'  => 'red',
+							'color' => 'red',
+						),
+						array(
+							'slug'  => 'green',
+							'color' => 'green',
+						),
```

Comment:
Non-blocking, but it looks like we caught several cases with manual testing in the refactor than with tests, what test cases may make sense to add?

Also wanted to note that as a reader with little context on the theme/color system, I have don't know what to expect in behavior from a test case named "test_merge_incoming_data"